Mirroring not updating

I've set up a mirror per the instructions at http://gitblit.com/setup_mirrors.html

The source repository is hosted on GitHub Enterprise, and it was mirrored using HTTPS with a username and password. The Gitblit log shows: (sanitized to not give internal user, repo, or hostnames)

```
2016-04-15 00:02:50 [ERROR] Error updating mirror problematic-repository.git
org.eclipse.jgit.api.errors.TransportException: https://gituser@internal.github.enterprise/ORG/problematic-repository.git: Authentication is required but no CredentialsProvider has been registered
        at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:139)
        at com.gitblit.service.MirrorService.run(MirrorService.java:149)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.eclipse.jgit.errors.TransportException: https://gituser@internal.github.enterprise/ORG/problematic-repository.git: Authentication is required but no CredentialsProvider has been registered
        at org.eclipse.jgit.transport.TransportHttp.connect(TransportHttp.java:492)
        at org.eclipse.jgit.transport.TransportHttp.openFetch(TransportHttp.java:296)
        at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:136)
        at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:122)
        at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1138)
        at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:130)
        ... 8 more
```

I notice that the logged URL does not include the password, but it is displayed in the Gitblit Web UI. I've been trying to follow exactly where and how the password may be lost (and hoping it's not a wild goose chase where it's simply not logged…), but I'm not succeeding so far. I'll update this ticket with code, if I manage to solve it on my own.

This is a showstopper bug for me, so I'll be investigating it heavily. However, if you have any suggestions for likely places to look (or know that it's fixed since 1.7.1), I'm all ears. 👂

--- a/src/main/java/com/gitblit/service/MirrorService.java
+++ b/src/main/java/com/gitblit/service/MirrorService.java
@@ -27,11 +27,14 @@ import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.lib.RefUpdate.Result;
 import org.eclipse.jgit.lib.Repository;
 import org.eclipse.jgit.lib.StoredConfig;
+import org.eclipse.jgit.transport.CredentialsProvider;
 import org.eclipse.jgit.transport.FetchResult;
 import org.eclipse.jgit.transport.ReceiveCommand;
 import org.eclipse.jgit.transport.ReceiveCommand.Type;
 import org.eclipse.jgit.transport.RemoteConfig;
 import org.eclipse.jgit.transport.TrackingRefUpdate;
+import org.eclipse.jgit.transport.URIish;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -146,7 +149,12 @@ public class MirrorService implements Runnable {
 				logger.debug("checking {} remote {} for ref updates", repositoryName, mirror.getName());
 				final boolean testing = false;
 				Git git = new Git(repository);
-				FetchResult result = git.fetch().setRemote(mirror.getName()).setDryRun(testing).call();
+				CredentialsProvider creds = null;
+				URIish fetchUri = mirror.getURIs().get(0);
+				if (fetchUri.getUser() != null && fetchUri.getPass() != null) {
+				    creds = new UsernamePasswordCredentialsProvider(fetchUri.getUser(), fetchUri.getPass());
+				}
+				FetchResult result = git.fetch().setCredentialsProvider(creds).setRemote(mirror.getName()).setDryRun(testing).call();
 				Collection<TrackingRefUpdate> refUpdates = result.getTrackingRefUpdates();
 				if (refUpdates.size() > 0) {
 					ReceiveCommand ticketBranchCmd = null;

